File: Scorekeeper/app/src/main/java/com/example/scorekeeper/MainActivity.java
package com.example.scorekeeper;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private int mScore1;
    private int mScore2;

    private TextView mScoreText1;
    private TextView mScoreText2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mScoreText1 = findViewById(R.id.score_1);
        mScoreText2 = findViewById(R.id.score_2);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    public void decreaseScore(View view) {
        int viewID = view.getId();

        switch (viewID) {
            case R.id.decreaseTeam1:
                mScore1--;

                if (mScore1 < 0) {
                    mScore1 = 0;
                    mScoreText1.setText(R.string.initial_score);
                } else {
                    mScoreText1.setText(String.valueOf(mScore1));
                }

                break;

            case R.id.decreaseTeam2:
                mScore2--;

                if (mScore2 < 0) {
                    mScore2 = 0;
                    mScoreText2.setText(R.string.initial_score);
                } else {
                    mScoreText2.setText(String.valueOf(mScore2));
                }

                break;
        }
    }

    public void increaseScore(View view) {
        int viewID = view.getId();

        switch (viewID) {
            case R.id.increaseTeam1:
                mScore1++;
                mScoreText1.setText(String.valueOf(mScore1));
                break;

            case R.id.increaseTeam2:
                mScore2++;
                mScoreText2.setText(String.valueOf(mScore2));
                break;
        }
    }
}

File: Scorekeeper/app/src/main/java/com/example/scorekeeper/MainActivity.kt
package com.example.scorekeeper

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate

class MainActivity : AppCompatActivity() {
    private var mScore1 = 0
    private var mScore2 = 0
    private var mScoreText1: TextView? = null
    private var mScoreText2: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mScoreText1 = findViewById(R.id.score_1)
        mScoreText2 = findViewById(R.id.score_2)

        if (savedInstanceState != null) {
            mScore1 = savedInstanceState.getInt(STATE_SCORE_1)
            mScore2 = savedInstanceState.getInt(STATE_SCORE_2)

            mScoreText1?.text = mScore1.toString()
            mScoreText2?.text = mScore2.toString()
        }
    }

    fun decreaseScore(view: View) {
        val viewID = view.id
        when (viewID) {
            R.id.decreaseTeam1 -> {
                mScore1--

                if (mScore1 < 0) {
                    mScore1 = 0
                    mScoreText1!!.setText(R.string.initial_score)
                } else {
                    mScoreText1!!.text = mScore1.toString()
                }
            }
            R.id.decreaseTeam2 -> {
                mScore2--

                if (mScore2 < 0) {
                    mScore2 = 0
                    mScoreText2!!.setText(R.string.initial_score)
                } else {
                    mScoreText2!!.text = mScore2.toString()
                }
            }
        }
    }

    fun increaseScore(view: View) {
        val viewID = view.id

        when (viewID) {
            R.id.increaseTeam1 -> {
                mScore1++
                mScoreText1!!.text = mScore1.toString()
            }
            R.id.increaseTeam2 -> {
                mScore2++
                mScoreText2!!.text = mScore2.toString()
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        val nightMode = AppCompatDelegate.getDefaultNightMode()

        if (nightMode == AppCompatDelegate.MODE_NIGHT_YES) {
            menu.findItem(R.id.night_mode).setTitle(R.string.day_mode)
        } else {
            menu.findItem(R.id.night_mode).setTitle(R.string.night_mode)
        }

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.night_mode) {
            val nightMode = AppCompatDelegate.getDefaultNightMode()

            if (nightMode == AppCompatDelegate.MODE_NIGHT_YES) {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
            } else {
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
            }

            // Recreate activity for changes to take effect
            recreate()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putInt(STATE_SCORE_1, mScore1)
        outState.putInt(STATE_SCORE_2, mScore2)
        super.onSaveInstanceState(outState)
    }

    companion object {
        const val STATE_SCORE_1 = "Team 1 Score"
        const val STATE_SCORE_2 = "Team 2 Score"
    }
}